params/hooks_libevm.go

```diff
@@ -142,11 +142,14 @@ func makePrecompile(contract contract.StatefulPrecompiledContract) libevm.Precom
 		}
 
 		callType := env.IncomingCallType()
-		isDissallowedCallType := callType == vm.DelegateCall || callType == vm.CallCode
-		if env.BlockTime() >= invalidateDelegateUnix {
-			if isDissallowedCallType {
-				env.InvalidateExecution(fmt.Errorf("precompile cannot be called with %s", callType))
+		isDisallowedCallType := callType == vm.DelegateCall || callType == vm.CallCode
+		if env.BlockTime() >= invalidateDelegateUnix && isDisallowedCallType {
+			if rules.IsGranite {
+				// After Granite activation, revert the transaction (include with failed receipt)
+				return nil, 0, vm.ErrExecutionReverted
 			}
+			// Before Granite activation, invalidate execution to exclude the tx from the block
+			env.InvalidateExecution(fmt.Errorf("precompile cannot be called with %s", callType))
```

Comment:
we should not be needing checking both rule activations
```suggestion
		if isDisallowedCallType {
			switch {
			case rules.IsGranite:
				// After Granite activation, revert the transaction (include with failed receipt)
				return nil, 0, vm.ErrExecutionReverted
			case env.BlockTime() >= invalidateDelegateUnix:
				// After expiration, invalidate execution to exclude the tx from the block
				env.InvalidateExecution(fmt.Errorf("precompile cannot be called with %s", callType))
			}
		}
```